### PR TITLE
[CELEBORN-1021] Celeborn support arbitary Ratis configs and client rpc timeout

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. To support arbitrary Ratis configs
2. To support Ratis client rpc timeout

### Why are the changes needed?
After some digs that I found out Celeborn never changed the default config of ratis client's timeout.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA and cluster.

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1774,7 +1779,23 @@ object CelebornConf extends Logging {
...
+      .timeConf(TimeUnit.SECONDS)
+      .createWithDefaultString("30s")
+
+  val HA_MASTER_RATIS_CLIENT_RPC_WATCH_TIMEOUT: ConfigEntry[Long] =
+    buildConf("celeborn.master.ha.ratis.raft.client.rpc.watch.timeout")
+      .internal
+      .categories("ha")
+      .version("0.3.2")
+      .timeConf(TimeUnit.SECONDS)
+      .createWithDefaultString("45s")
```
Comment (cxzl25) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1798]: Looks larger than the original default value.

ratis client default config
```properties
raft.client.rpc.request.timeout=3000ms
raft.client.rpc..watch.request.timeout=10000ms
```

https://github.com/apache/ratis/blob/2e4e75bdc9671d9ac03eae50ce02ff4aa48ced1f/ratis-client/src/main/java/org/apache/ratis/client/RaftClientConfigKeys.java#L44-L56

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1798]: Yes, setting this config larger than the default value is on purpose.

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1798]: OK, docs will be updated in this PR.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -1776,6 +1781,22 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.SECONDS)
       .createWithDefaultString("5s")
 
+  val HA_MASTER_RATIS_CLIENT_RPC_TIMEOUT: ConfigEntry[Long] =
+    buildConf("celeborn.master.ha.ratis.raft.client.rpc.timeout")
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1785]: Could this add method to parse the config of Ratis itself, which could not alway follow the Ratis config change?

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1785]: Yes, there is a way to allow users who are familiar with Ratis to pass all their own Ratis configs to Celeborn.

**Review by RexXiong — APPROVED** [review]

**Pull request closed without merging**